packages/keyring-controller/src/KeyringController.ts

```diff
@@ -218,6 +220,17 @@ export class KeyringController extends BaseController<
       throw new Error('No HD keyring found');
     }
     const oldAccounts = await this.#keyring.getAccounts();
+
+    if (oldAccounts.length !== accountCount) {
+      // we return the last account added to the primary keyring
+      const primaryKeyringAccounts = await primaryKeyring.getAccounts();
+      return {
+        keyringState: await this.fullUpdate(),
+        addedAccountAddress:
+          primaryKeyringAccounts[primaryKeyringAccounts.length - 1],
```

Comment:
This risks being non-idempotent if another call is made in-between, e.g. imagine two calls are made, and get queued up together:

```
addNewAccount(10); // returns 11th account
addNewAccount(11); // returns 12th account
addNewAccount(10); // returns 12th account
addNewAccount(11); // returns 12th account
```

This is a concern the extension didn't have because of the lack of an "addedAccountAddress" return parameter.

Maybe something like this would work?

```suggestion
      // we return the last account added to the primary keyring
      const primaryKeyringAccounts = await primaryKeyring.getAccounts();
      return {
        keyringState: await this.fullUpdate(),
        addedAccountAddress:
          primaryKeyringAccounts[accountCount],
```